internal/controller/infrastructure/provisioner.go

```diff
@@ -98,6 +104,56 @@ func (p *Provisioner) Provision(_ context.Context) error {
 	return nil
 }
 
+// Cleanup cleans up a machine
+// The provisioning process is as follows:
+// 1. Open SSH connection to the machine
+// 2. Stops k0s
+// 3. Removes node from etcd
+// 4. Runs k0s reset
+func (p *Provisioner) Cleanup(_ context.Context, mode RemoteMachineMode) error {
+	if mode == ModeNonK0s {
+		return nil
+	}
+
+	authM, err := rig.ParseSSHPrivateKey(p.sshKey, rig.DefaultPasswordCallback)
```

Comment:
I don't know if it's a problem here but the password callback will prompt for input if the key is passphrase protected. If you want to avoid having it waiting for input forever, just pass nil.

Reply:
Thanks!